test/langtools/tools/javac/lambda/deduplication/Deduplication.java

```diff
@@ -187,6 +187,9 @@ void f() {}
                     int y2 = -1;
                     return y2;
                });
+
+        group((Function<Object, Integer>) x -> {class C {} new C(); return 42; });
```

Comment:
It feels like the test logic could be maybe cleaner if `group` had two variants `groupEquals` and `groupNotEquals` - e.g. in this case you want to make sure the lambdas in the group do NOT dedup.